importer/tests/writer_test.py

```diff
@@ -27,9 +30,54 @@ def test_copy_files_single_strand(self, copyfile_patch):
             RawRead(forward_read='PAIR1_1.fastq.gz', reverse_read='PAIR1_2.fastq.gz', sample_name='SAMPLE1',
                     taxon_id='1280', library_name='LIB1', sample_accession=None),
             RawRead(forward_read='SINGLE.fastq.gz', reverse_read=None, sample_name='SAMPLE2',
-                    taxon_id='1280', library_name='LIB2', sample_accession=None)]), 'destination', 0)
+                    taxon_id='1280', library_name='LIB2', sample_accession=None)]), 'destination', 0, 0)
         under_test.copy_files('source')
         self.assertEquals(copyfile_patch.call_args_list,
                           [call('source/PAIR1_1.fastq.gz', 'destination/0/PAIR1_1.fastq.gz'),
                            call('source/PAIR1_2.fastq.gz', 'destination/0/PAIR1_2.fastq.gz'),
                            call('source/SINGLE.fastq.gz', 'destination/0/SINGLE.fastq.gz')])
+
+
+class TestXlsGeneration(unittest.TestCase):
+
+    def test_of_OutputSpreadsheetGenerator(self):
+        sheet = Spreadsheet()
+        sheet.supplier, sheet.organisation, sheet.contact, sheet.technology, sheet.name, sheet.accession, sheet.size,\
+        sheet.limit = ('Supplier', 'Org', 'Contact', 'Illumina', 'AStudyName1', None, 1.90, '01/01/2025')
+        sheet.reads = [RawRead(forward_read='PAIR1_1.fastq.gz', reverse_read='PAIR1_2.fastq.gz', sample_name='SAMPLE1',
+                              taxon_id='1280', library_name='LIB1', sample_accession=None),
+                       RawRead(forward_read='SECONDLINE.fastq.gz', reverse_read=None, sample_name='SAMPLE2',
+                              taxon_id='1280', library_name='LIB2', sample_accession=None)]
+        current_position = 1
+        breakpoint = 1
+        generator = OutputSpreadsheetGenerator(sheet, current_position)
+        workbook, file_ended, current_position = generator.build(breakpoint)
+        workbook.save('workbook.xls')
+        self.assertEqual((current_position, file_ended), (2, True))
+        please = xlrd.open_workbook('workbook.xls')
```

Comment:
I'd create a dedicated function/method to validate sheet content.  It will be easier to read, but also will allow you to name the variable properly without conflict